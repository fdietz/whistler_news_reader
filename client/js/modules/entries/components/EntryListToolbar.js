import React, { PropTypes } from 'react';
import classNames from 'classnames';

import Icon from '../../../components/Icon';
import {
  CheckmarkSVGIcon,
  TrashSVGIcon,
  CycleSVGIcon,
  CogSVGIcon,
  ArrowDownSVGIcon,
  ArrowUpSVGIcon,
  MenuSVGIcon,
  EllipsisSVGIcon
} from '../../../components/SVGIcon';

import Button from '../../../components/Button';
import ButtonGroup from '../../../components/ButtonGroup';
import DropdownTrigger from '../../../components/DropdownTrigger';
import DropdownContent from '../../../components/DropdownContent';
import Dropdown from '../../../components/Dropdown';

const EntryListToolbar = ({
  isMobile,
  title,
  currentViewLayout,
  showSpinner,
  isSubscriptionSelected,
  isCategorySelected,
  onMarkAsReadClick,
  onRefreshEntriesClick,
  onRemoveFeedOrCategoryClick,
  onViewLayoutChangeClick,
  onOpenEditFeedOrCategoryModalClick,
  onToggleSidebarClick,
}) => {
  const spinnerCls = classNames('spinner spinner-blink', {
    hide: !showSpinner,
  });

  const editFeedOrCategoryCls = classNames('dropdown__list-item', {
    disabled: !isSubscriptionSelected && !isCategorySelected,
  });

  const toolbarCls = classNames('toolbar', {
    'toolbar--titled': isMobile,
    'show-small-screen-only': isMobile,
    'hide-small-screen': !isMobile
  });

  const dropdownCls = classNames('dropdown', {
    west: isMobile
  });

  return (
    <div className={toolbarCls}>
      {isMobile &&
        <Button
          type="header"
          onClick={onToggleSidebarClick}
          title="Toggle Sidebar"
          className="btn-logo">
          <MenuSVGIcon color="#F25653" size="small" />
        </Button>
      }
      {isMobile &&
        <header>
          <div className="title">{title}</div>
          <span className={spinnerCls}>
            <Icon name="spinner_white" size="small" />
          </span>
        </header>
      }

      {!isMobile &&
        <Button
          type="header"
          onClick={onToggleSidebarClick}
          title="Mark all stories as read"
          className="hide-large-screen mr1-important">
          <MenuSVGIcon color="#F25653" size="small" />
        </Button>
      }
      {!isMobile &&
        <ButtonGroup className="btn-group-rounded">
          <Button
            type="header"
            onClick={onMarkAsReadClick}
            title="Mark all stories as read">
            <CheckmarkSVGIcon color="light-gray" size="small" />
          </Button>
          <Button
            type="header"
            onClick={onRefreshEntriesClick}
            title="Fetch new stories">
            <CycleSVGIcon color="light-gray" size="small" />
          </Button>
        </ButtonGroup>
      }

      <div className="right">
        {isMobile &&
          <Button
            type="header"
            onClick={onRefreshEntriesClick}
            title="Fetch new stories">
            <CycleSVGIcon color="light-gray" size="small" />
          </Button>
        }
        <Dropdown className={dropdownCls}>
          <DropdownTrigger className="btn btn-header">
            {isMobile && <EllipsisSVGIcon color="light-gray" size="small" />}
            {!isMobile && <CogSVGIcon color="light-gray" size="small" />}
            {!isMobile && <ArrowDownSVGIcon color="light-gray" size="small" className="arrow-down" />}
            {!isMobile && <ArrowUpSVGIcon color="light-gray" size="small" className="arrow-up" />}
          </DropdownTrigger>

          <DropdownContent>
            <ul className="dropdown__list">
              {isMobile &&
                <li className="dropdown__list-item">
                  <a
                    onClick={onMarkAsReadClick}
                    title="Mark all stories as read">
                    <CheckmarkSVGIcon color="light-gray" size="small" />
                    Mark all as read
                  </a>
                </li>
              }
              {isMobile &&
                <li className="dropdown__list-separator" />
              }

              <li
                className="dropdown__list-item"
                onClick={onViewLayoutChangeClick.bind(this, 'list')}>
                <input
                  type="radio"
                  id="list"
                  name="currrent_view_layout"
                  checked={currentViewLayout === 'list'}
                  readOnly
                  className="dropdown-field media" />
                <label
                  htmlFor="list"
                  className="dropdown-label content" >List</label>
              </li>
              <li
                className="dropdown__list-item"
                onClick={onViewLayoutChangeClick.bind(this, 'compact_list')}>
                <input
                  type="radio"
                  id="compact_list"
                  name="currrent_view_layout"
                  checked={currentViewLayout === 'compact_list'}
                  readOnly
                  className="dropdown-field media" />
                <label
                  htmlFor="compact_list"
                  className="dropdown-label
                  content">Compact List</label>
              </li>
              <li
                className="dropdown__list-item"
                onClick={onViewLayoutChangeClick.bind(this, 'grid')}>
                <input
                  type="radio"
                  id="grid"
                  name="currrent_view_layout"
                  checked={currentViewLayout === 'grid'}
                  readOnly
                  className="dropdown-field media" />
                <label
                  htmlFor="grid"
                  className="dropdown-label content">Grid (experimental)</label>
              </li>
              <li className="dropdown__list-separator" />
              <li
                className={editFeedOrCategoryCls}
                onClick={onOpenEditFeedOrCategoryModalClick}>
                <div className="media">
                  <CogSVGIcon color="light-gray" size="small" />
                </div>
                <div className="content">Settings</div>
              </li>
              <li
                className={editFeedOrCategoryCls}
                onClick={onRemoveFeedOrCategoryClick}>
                <div className="media">
                  <TrashSVGIcon color="light-gray" size="small" />
                </div>
                <div className="content">Delete</div>
              </li>
            </ul>
          </DropdownContent>
        </Dropdown>
      </div>
      <span className={spinnerCls}>
        <Icon name="spinner" size="small" />
      </span>
    </div>
  );
};

EntryListToolbar.propTypes = {
  isMobile: PropTypes.bool.isRequired,
  title: PropTypes.string.isRequired,
  currentViewLayout: PropTypes.string.isRequired,
  showSpinner: PropTypes.bool.isRequired,
  isSubscriptionSelected: PropTypes.bool.isRequired,
  isCategorySelected: PropTypes.bool.isRequired,
  hasPreviousEntry: PropTypes.bool.isRequired,
  hasNextEntry: PropTypes.bool.isRequired,
  onMarkAsReadClick: PropTypes.func.isRequired,
  onRefreshEntriesClick: PropTypes.func.isRequired,
  onRemoveFeedOrCategoryClick: PropTypes.func.isRequired,
  onViewLayoutChangeClick: PropTypes.func.isRequired,
  onOpenEditFeedOrCategoryModalClick: PropTypes.func.isRequired,
  onPreviousEntryClick: PropTypes.func.isRequired,
  onNextEntryClick: PropTypes.func.isRequired,
  onOpenExternalClick: PropTypes.func.isRequired,
  onToggleSidebarClick: PropTypes.func.isRequired
};

export default EntryListToolbar;
