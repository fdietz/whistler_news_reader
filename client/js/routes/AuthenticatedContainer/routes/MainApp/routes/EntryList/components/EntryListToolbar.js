import React, { PropTypes } from 'react';
import classNames from 'classnames';

import Icon from 'components/Icon';
import { CheckmarkSVGIcon, TrashSVGIcon, CycleSVGIcon, CogSVGIcon,
  ArrowDownSVGIcon, ArrowUpSVGIcon, MenuSVGIcon } from 'components/SVGIcon';

import Button from 'components/Button';
import ButtonGroup from 'components/ButtonGroup';
import DropdownTrigger from 'components/DropdownTrigger';
import DropdownContent from 'components/DropdownContent';
import Dropdown from 'components/Dropdown';

const EntryListToolbar = ({
  currentViewLayout,
  showSpinner,
  isSubscriptionSelected,
  isCategorySelected,
  onMarkAsReadClick,
  onRefreshEntriesClick,
  onRemoveFeedOrCategoryClick,
  onViewLayoutChangeClick,
  onOpenEditFeedOrCategoryModalClick,
  onToggleSidebarClick,
}) => {
  const spinnerCls = classNames('spinner spinner-blink', {
    hide: !showSpinner,
  });

  const editFeedOrCategoryCls = classNames('dropdown__list-item', {
    disabled: !isSubscriptionSelected && !isCategorySelected,
  });

  return (
    <div className="toolbar hide-small-screen">
      <Button
        type="header"
        onClick={onToggleSidebarClick}
        title="Mark all stories as read"
        className="hide-large-screen mr1-important">
        <MenuSVGIcon color="#F25653" size="small" />
      </Button>
      <ButtonGroup className="btn-group-rounded">
        <Button
          type="header"
          onClick={onMarkAsReadClick}
          title="Mark all stories as read">
          <CheckmarkSVGIcon color="light-gray" size="small" />
        </Button>
        <Button
          type="header"
          onClick={onRefreshEntriesClick}
          title="Fetch new stories">
          <CycleSVGIcon color="light-gray" size="small" />
        </Button>
      </ButtonGroup>
      <Dropdown className="ml1">
        <DropdownTrigger className="btn btn-header">
          <CogSVGIcon color="light-gray" size="small" />
          <ArrowDownSVGIcon color="light-gray" size="small" className="arrow-down" />
          <ArrowUpSVGIcon color="light-gray" size="small" className="arrow-up" />
        </DropdownTrigger>
        <DropdownContent>
          <ul className="dropdown__list">
            <li
              className="dropdown__list-item"
              onClick={onViewLayoutChangeClick.bind(this, 'list')}>
              <input
                type="radio"
                id="list"
                name="currrent_view_layout"
                checked={currentViewLayout === 'list'}
                readOnly
                className="dropdown-field media" />
              <label
                htmlFor="list"
                className="dropdown-label content" >List</label>
            </li>
            <li
              className="dropdown__list-item"
              onClick={onViewLayoutChangeClick.bind(this, 'compact_list')}>
              <input
                type="radio"
                id="compact_list"
                name="currrent_view_layout"
                checked={currentViewLayout === 'compact_list'}
                readOnly
                className="dropdown-field media" />
              <label
                htmlFor="compact_list"
                className="dropdown-label
                content">Compact List</label>
            </li>
            <li
              className="dropdown__list-item"
              onClick={onViewLayoutChangeClick.bind(this, 'grid')}>
              <input
                type="radio"
                id="grid"
                name="currrent_view_layout"
                checked={currentViewLayout === 'grid'}
                readOnly
                className="dropdown-field media" />
              <label
                htmlFor="grid"
                className="dropdown-label content">Grid (experimental)</label>
            </li>
            <li className="dropdown__list-separator" />
            <li
              className={editFeedOrCategoryCls}
              onClick={onOpenEditFeedOrCategoryModalClick}>
              <div className="media">
                <CogSVGIcon color="light-gray" size="small" />
              </div>
              <div className="content">Settings</div>
            </li>
            <li
              className={editFeedOrCategoryCls}
              onClick={onRemoveFeedOrCategoryClick}>
              <div className="media">
                <TrashSVGIcon color="light-gray" size="small" />
              </div>
              <div className="content">Delete</div>
            </li>
          </ul>
        </DropdownContent>
      </Dropdown>
      <span className={spinnerCls}>
        <Icon name="spinner" size="small" />
      </span>
    </div>
  );
};

EntryListToolbar.propTypes = {
  currentViewLayout: PropTypes.string.isRequired,
  showSpinner: PropTypes.bool.isRequired,
  isSubscriptionSelected: PropTypes.bool.isRequired,
  isCategorySelected: PropTypes.bool.isRequired,
  hasPreviousEntry: PropTypes.bool.isRequired,
  hasNextEntry: PropTypes.bool.isRequired,
  onMarkAsReadClick: PropTypes.func.isRequired,
  onRefreshEntriesClick: PropTypes.func.isRequired,
  onRemoveFeedOrCategoryClick: PropTypes.func.isRequired,
  onViewLayoutChangeClick: PropTypes.func.isRequired,
  onOpenEditFeedOrCategoryModalClick: PropTypes.func.isRequired,
  onPreviousEntryClick: PropTypes.func.isRequired,
  onNextEntryClick: PropTypes.func.isRequired,
  onOpenExternalClick: PropTypes.func.isRequired,
};

export default EntryListToolbar;
