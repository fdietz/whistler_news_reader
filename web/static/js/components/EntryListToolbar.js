import React, { PropTypes } from "react";
import classNames from "classnames";

import Icon from "../components/Icon";
import { CheckmarkSVGIcon, TrashSVGIcon, CycleSVGIcon, ArrowLeftBoldSVGIcon,
  ArrowRightBoldSVGIcon, EarthSVGIcon, ResizeEnlargeSVGIcon, CogSVGIcon,
  ArrowDownSVGIcon, ArrowUpSVGIcon } from "../components/SVGIcon";

import Button from "../components/Button";
import ButtonGroup from "../components/ButtonGroup";
import DropdownTrigger from "../components/DropdownTrigger";
import DropdownContent from "../components/DropdownContent";
import Dropdown from "../components/Dropdown";

const EntryListToolbar = ({
  currentViewLayout,
  entries,
  currentSidebarSelection,
  onMarkAsReadClick,
  onRefreshEntriesClick,
  onRemoveFeedOrCategoryClick,
  onViewLayoutChangeClick,
  onPreviousEntryClick,
  onNextEntryClick,
  onOpenExternalClick,
  onOpenEntryContentModalClick,
  onOpenEditFeedOrCategoryModalClick
}) => {
  const spinnerCls = classNames({
    spinner: true,
    hide: !entries.isLoading
  });

  const isFeedOrCategorySelected = !currentSidebarSelection.selection;
  const editFeedOrCategoryCls = classNames("dropdown__list-item", {
    disabled: isFeedOrCategorySelected
  });

  return (
  <div className="toolbar">
    <ButtonGroup className="btn-group-rounded">
        <Button
          type="header"
          onClick={onMarkAsReadClick}
          title="Mark all stories as read">
          <CheckmarkSVGIcon color="light-gray" size="small"/>
        </Button>
        <Button
          type="header"
          onClick={onRefreshEntriesClick}
          title="Fetch new stories">
          <CycleSVGIcon color="light-gray" size="small"/>
        </Button>
      </ButtonGroup>
      <Dropdown className="ml1">
        <DropdownTrigger className="btn btn-header">
          <CogSVGIcon color="light-gray" size="small"/>
          <ArrowDownSVGIcon color="light-gray" size="small" className="arrow-down"/>
          <ArrowUpSVGIcon color="light-gray" size="small" className="arrow-up"/>
        </DropdownTrigger>
        <DropdownContent>
          <ul className="dropdown__list">
            <li
              className="dropdown__list-item"
              onClick={onViewLayoutChangeClick.bind(this, "list")}>
                <input
                  type="radio"
                  id="list"
                  name="currrent_view_layout"
                  checked={currentViewLayout === "list"}
                  className="dropdown-field media"/>
                <label
                  htmlFor="list"
                  className="dropdown-label content">List</label>
            </li>
            <li
              className="dropdown__list-item"
              onClick={onViewLayoutChangeClick.bind(this, "compact_list")}>
                <input
                  type="radio"
                  id="compact_list"
                  name="currrent_view_layout"
                  checked={currentViewLayout === "compact_list"}
                  className="dropdown-field media"/>
                <label
                  htmlFor="compact_list"
                  className="dropdown-label
                  content">Compact List</label>
            </li>
            <li
              className="dropdown__list-item"
              onClick={onViewLayoutChangeClick.bind(this, "grid")}>
                <input
                  type="radio"
                  id="grid"
                  name="currrent_view_layout"
                  checked={currentViewLayout === "grid"}
                  className="dropdown-field media"/>
                <label
                  htmlFor="grid"
                  className="dropdown-label content">Grid</label>
            </li>
            <li className="dropdown__list-separator"/>
            <li
              className={editFeedOrCategoryCls}
              onClick={onOpenEditFeedOrCategoryModalClick}>
              <div className="media">
                <CogSVGIcon color="light-gray" size="small"/>
              </div>
              <div className="content">Settings</div>
            </li>
            <li
              className={editFeedOrCategoryCls}
              onClick={onRemoveFeedOrCategoryClick}>
              <div className="media">
                <TrashSVGIcon color="light-gray" size="small"/>
              </div>
              <div className="content">Delete</div>
            </li>
          </ul>
        </DropdownContent>
      </Dropdown>
      {currentViewLayout === "grid" &&
        <ButtonGroup className="btn-group-rounded ml2">
          <Button type="header" onClick={onPreviousEntryClick}>
            <ArrowLeftBoldSVGIcon color="light-gray" size="small"/>
          </Button>
          <Button type="header" onClick={onNextEntryClick}>
            <ArrowRightBoldSVGIcon color="light-gray" size="small"/>
          </Button>
        </ButtonGroup>
      }
      {currentViewLayout === "grid" &&
        <ButtonGroup className="btn-group-rounded ml2">
          <Button type="header" onClick={onOpenEntryContentModalClick}>
            <ResizeEnlargeSVGIcon color="light-gray" size="small"/>
          </Button>
          <Button type="header" onClick={onOpenExternalClick}>
            <EarthSVGIcon color="light-gray" size="small"/>
          </Button>
        </ButtonGroup>
      }
      <span className={spinnerCls}>
        <Icon name="spinner" size="small"/>
      </span>
    </div>
  );
};

EntryListToolbar.propTypes = {
  currentViewLayout: PropTypes.string.isRequired,
  entries: PropTypes.object.isRequired,
  currentSidebarSelection: PropTypes.object.isRequired,
  onMarkAsReadClick: PropTypes.func.isRequired,
  onRefreshEntriesClick: PropTypes.func.isRequired,
  onRemoveFeedOrCategoryClick: PropTypes.func.isRequired,
  onViewLayoutChangeClick: PropTypes.func.isRequired,
  onOpenEditFeedOrCategoryModalClick: PropTypes.func.isRequired,
  onPreviousEntryClick: PropTypes.func.isRequired,
  onNextEntryClick: PropTypes.func.isRequired,
  onOpenExternalClick: PropTypes.func.isRequired,
  onOpenEntryContentModalClick: PropTypes.func.isRequired
};

export default EntryListToolbar;
