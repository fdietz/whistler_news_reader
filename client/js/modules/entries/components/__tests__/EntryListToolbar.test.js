import renderer from 'react-test-renderer';
import React from 'react';
import EntryListToolbar from '../EntryListToolbar';

describe('EntryListToolbar', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <EntryListToolbar
        isMobile={true}
        title="title"
        currentViewLayout="normal"
        showSpinner={false}
        isSubscriptionSelected={true}
        isCategorySelected={true}
        onMarkAsReadClick={() => {}}
        onRefreshEntriesClick={() => {}}
        onRemoveFeedOrCategoryClick={() => {}}
        onViewLayoutChangeClick={() => {}}
        onOpenEditFeedOrCategoryModalClick={() => {}}
        onToggleSidebarClick={() => {}}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
