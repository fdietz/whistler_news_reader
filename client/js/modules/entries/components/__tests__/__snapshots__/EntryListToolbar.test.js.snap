exports[`EntryListToolbar renders correctly 1`] = `
<div
  className="toolbar toolbar--titled show-small-screen-only">
  <button
    className="btn logo btn-header"
    onClick={[Function onToggleSidebarClick]}
    title="Toggle Sidebar">
    <svg
      height={20}
      viewBox="0 0 20 20 "
      width={20}>
      <path
        d="M1.25 3.75h17.5v3.75h-17.5zM1.25 8.75h17.5v3.75h-17.5zM1.25 13.75h17.5v3.75h-17.5z"
        fill="#98B0D2" />
    </svg>
  </button>
  <header>
    <div
      className="title">
      title
    </div>
    <span
      className="spinner spinner-blink hide">
      <span
        className="svg-entypo-icon-spinner_white svg-icon-small" />
    </span>
  </header>
  <div
    className="right">
    <button
      className="btn btn-header"
      onClick={[Function onRefreshEntriesClick]}
      title="Fetch new stories">
      <svg
        height={20}
        viewBox="0 0 20 20 "
        width={20}>
        <path
          d="M5.516 14.224c-2.262-2.433-2.222-6.245 0.127-8.611 0.962-0.968 2.164-1.546 3.414-1.736l-0.069-2.077c-1.754 0.213-3.452 0.996-4.797 2.35-3.149 3.172-3.186 8.29-0.123 11.532l-1.741 1.753 5.51 0.301-0.015-5.834-2.306 2.322zM12.162 2.265l0.015 5.834 2.307-2.323c2.262 2.433 2.223 6.245-0.127 8.611-0.962 0.969-2.164 1.546-3.415 1.736l0.069 2.076c1.755-0.213 3.453-0.996 4.798-2.351 3.148-3.171 3.187-8.29 0.122-11.531l1.741-1.753-5.51-0.299z"
          fill="#98B0D2" />
      </svg>
    </button>
    <div
      className="dropdown dropdown west"
      style={undefined}>
      <a
        className="dropdown__trigger btn btn-header"
        onClick={[Function bound toggleActive]}
        style={undefined}>
        <svg
          height={20}
          viewBox="0 0 20 20 "
          width={20}>
          <path
            d="M3 7.8c-1.215 0-2.2 0.985-2.2 2.2s0.985 2.2 2.2 2.2 2.2-0.985 2.2-2.2c0-1.215-0.985-2.2-2.2-2.2zM10 7.8c-1.215 0-2.2 0.985-2.2 2.2s0.985 2.2 2.2 2.2 2.199-0.985 2.199-2.2c0-1.215-0.984-2.2-2.199-2.2zM17 7.8c-1.215 0-2.2 0.985-2.2 2.2s0.985 2.2 2.2 2.2 2.2-0.985 2.2-2.2c0-1.215-0.985-2.2-2.2-2.2z"
            fill="#98B0D2" />
        </svg>
      </a>
      <div
        className="dropdown__content hidden"
        onClick={[Function bound handleClickInsideDropdownContent]}
        style={undefined}>
        <ul
          className="dropdown__list">
          <li
            className="dropdown__list-item">
            <a
              onClick={[Function onMarkAsReadClick]}
              title="Mark all stories as read">
              <svg
                height={20}
                viewBox="0 0 20 20 "
                width={20}>
                <path
                  d="M8.295 17c-0.435 0-0.847-0.203-1.111-0.553l-3.574-4.718c-0.465-0.614-0.344-1.488 0.27-1.953s1.488-0.344 1.953 0.27l2.351 3.104 5.912-9.491c0.407-0.653 1.267-0.853 1.921-0.446 0.653 0.406 0.853 1.267 0.445 1.92l-6.983 11.21c-0.243 0.391-0.662 0.636-1.121 0.656-0.021 0.001-0.042 0.001-0.063 0.001z"
                  fill="#98B0D2" />
              </svg>
              Mark all as read
            </a>
          </li>
          <li
            className="dropdown__list-separator" />
          <li
            className="dropdown__list-item"
            onClick={[Function bound onViewLayoutChangeClick]}>
            <input
              checked={false}
              className="dropdown-field media"
              id="list"
              name="currrent_view_layout"
              readOnly={true}
              type="radio" />
            <label
              className="dropdown-label content"
              htmlFor="list">
              List
            </label>
          </li>
          <li
            className="dropdown__list-item"
            onClick={[Function bound onViewLayoutChangeClick]}>
            <input
              checked={false}
              className="dropdown-field media"
              id="compact_list"
              name="currrent_view_layout"
              readOnly={true}
              type="radio" />
            <label
              className="dropdown-label content"
              htmlFor="compact_list">
              Compact List
            </label>
          </li>
          <li
            className="dropdown__list-item"
            onClick={[Function bound onViewLayoutChangeClick]}>
            <input
              checked={false}
              className="dropdown-field media"
              id="grid"
              name="currrent_view_layout"
              readOnly={true}
              type="radio" />
            <label
              className="dropdown-label content"
              htmlFor="grid">
              Grid (experimental)
            </label>
          </li>
          <li
            className="dropdown__list-separator" />
          <li
            className="dropdown__list-item"
            onClick={[Function onOpenEditFeedOrCategoryModalClick]}>
            <div
              className="media">
              <svg
                height={20}
                viewBox="0 0 20 20 "
                width={20}>
                <path
                  d="M16.785 10c0-1.048 0.645-1.875 1.615-2.443-0.176-0.584-0.406-1.145-0.691-1.672-1.090 0.285-1.971-0.142-2.711-0.882-0.742-0.741-0.969-1.623-0.684-2.711-0.527-0.285-1.086-0.518-1.672-0.692-0.568 0.971-1.595 1.616-2.642 1.616s-2.073-0.645-2.642-1.616c-0.586 0.174-1.145 0.407-1.672 0.691 0.284 1.089 0.058 1.971-0.683 2.712s-1.623 1.167-2.711 0.883c-0.285 0.526-0.517 1.087-0.692 1.671 0.971 0.568 1.616 1.395 1.616 2.443 0 1.047-0.645 2.074-1.616 2.643 0.175 0.585 0.407 1.145 0.692 1.672 1.089-0.285 1.97-0.058 2.711 0.683s0.967 1.622 0.683 2.711c0.527 0.285 1.087 0.518 1.672 0.692 0.569-0.972 1.595-1.616 2.642-1.616s2.075 0.645 2.643 1.616c0.586-0.175 1.145-0.407 1.672-0.692-0.285-1.089-0.059-1.97 0.684-2.711 0.74-0.74 1.621-1.167 2.711-0.883 0.285-0.527 0.516-1.087 0.691-1.672-0.971-0.569-1.616-1.396-1.616-2.443zM10 13.653c-2.017 0-3.652-1.636-3.652-3.653s1.635-3.653 3.652-3.653c2.018 0 3.653 1.636 3.653 3.653s-1.635 3.653-3.653 3.653z"
                  fill="#98B0D2" />
              </svg>
            </div>
            <div
              className="content">
              Settings
            </div>
          </li>
          <li
            className="dropdown__list-item"
            onClick={[Function onRemoveFeedOrCategoryClick]}>
            <div
              className="media">
              <svg
                height={20}
                viewBox="0 0 20 20 "
                width={20}>
                <path
                  d="M3.389 7.912l1.102 9.709c0.059 0.461 2.285 1.977 5.509 1.979 3.224-0.002 5.451-1.518 5.51-1.979l1.102-9.709c-1.684 0.942-4.201 1.388-6.612 1.388s-4.928-0.446-6.611-1.388zM13.167 1.912l-0.859-0.953c-0.331-0.471-0.691-0.559-1.392-0.559h-1.832c-0.7 0-1.060 0.088-1.392 0.559l-0.858 0.953c-2.57 0.449-4.434 1.64-4.434 2.518v0.171c0 1.546 3.402 2.8 7.6 2.8s7.6-1.254 7.6-2.8v-0.171c0-0.878-1.863-2.069-4.433-2.518zM12.069 4.139l-1.252-1.339h-1.634l-1.252 1.339h-1.7c0 0 1.862-2.22 2.111-2.521 0.19-0.23 0.385-0.318 0.637-0.318h2.043c0.252 0 0.446 0.088 0.637 0.318 0.248 0.301 2.111 2.521 2.111 2.521h-1.701z"
                  fill="#98B0D2" />
              </svg>
            </div>
            <div
              className="content">
              Delete
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
